File: Exercism/javascript/series/series.js
'use strict'

export class Series {
  constructor (string) {
    this.serie = [...string].map(Number)
  }

  get digits () {
    return this.serie
  }

  slices (n) {
    if (n > this.serie.length) {
      throw new Error('Slice size is too big.')
    }
    return this.serie
      .reduce((slcs, __, i, serie) =>
          n + i <= serie.length
            ? [...slcs, serie.slice(i, n + i)]
            : slcs
        , [])
  }
}

File: Exercism/python/series/series.py
def slices(series: str, length: int) -> list:
    """Given a string of digits, output all the contiguous substrings
    of length 'n' in that string in the order that they appear."""
    if length > len(series) or length <= 0:
        raise ValueError('Incorrect slice length')
    return [series[i:i+length] for i in range(len(series) - length + 1)]
